src/Microsoft.Diagnostics.Monitoring.EventPipe/Counters/TraceEventExtensions.cs

```diff
@@ -29,8 +29,23 @@ public CounterConfiguration(CounterFilter filter)
 
     internal static class TraceEventExtensions
     {
+        private static Dictionary<(string, string), CachedCounterInfo> counterInfoByName = new();
```

Comment:
You might need a custom comparer here to ensure that the keys are compared correctly. I get that this is a tuple, and tuples should handle comparison by default, but the arguments of the tuple are strings. The `Tuple<T1,T2>` class does implement `IComparable` and `IStructuralComparable` but it doesn't know to use the specialized comparer for the `string` properties.

If this were simply a `Dictionary<string, ...>`, I would be pointing out that you would need to specify a `StringComparer`.

So you might need to make this a formal class (a `record struct` should work well here) and then implement a comparer for that type that uses the appropriate `StringComparer` for each of the string properties.

Reply:
I haven't worked much with record structs before, so let me know if you want any more changes to what I did around this if I'm missing something.